spring-kafka/src/main/java/org/springframework/kafka/listener/AfterRollbackProcessor.java

```diff
@@ -40,10 +40,25 @@
 public interface AfterRollbackProcessor<K, V> {
 
 	/**
-	 * Process the remaining records.
+	 * Process the remaining records. Recoverable will be true if the container is
+	 * processing individual records; this allows the processor to recover (skip) the
+	 * failed record rather than re-seeking it. This is not possible with a batch listener
+	 * since only the listener itself knows which record in the batch keeps failing.
 	 * @param records the records.
 	 * @param consumer the consumer.
+	 * @param exception the exception
+	 * @param recoverable the recoverable.
+	 * @since 2.2
 	 */
-	void process(List<ConsumerRecord<K, V>> records, Consumer<K, V> consumer);
+	void process(List<ConsumerRecord<K, V>> records, Consumer<K, V> consumer, Exception exception, boolean recoverable);
```

Comment:
I don't think it's a big breaking change - as long as we add it to the migration guide - if I do the `default` dance, it can no longer be a `@FunctionalInterface`, so that would be a breaking change too.

I think custom implementations will be rare.

Reply:
If you have strong objections, I suppose I could create a new sub-interface.

Oops - I see you merged already.